metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/AbstractStrategy.java

```diff
@@ -539,12 +594,57 @@ protected void setJobDurationMetricInHostMonitor(
 				.append(connectorId)
 				.append("\"}")
 				.toString();
-			metricFactory.collectNumberMetric(
-				endpointHostMonitor,
-				jobDurationMetricKey,
-				(endTime - startTime) / 1000.0, // Job duration in seconds
-				strategyTime
-			);
+			// Collect the job duration metric
+			collectJobDurationMetric(jobDurationMetricKey, startTime, endTime);
+		}
+	}
+
+	/**
+	 * Sets the job duration metric in the host monitor without a monitor type.
+	 *
+	 * @param jobName      the name of the job
+	 * @param connectorId  the ID of the connector
+	 * @param startTime the start time of the job in milliseconds
+	 * @param endTime   the end time of the job in milliseconds
+	 */
+	protected void setJobDurationMetricInHostMonitorWithoutMonitorType(
```

Comment:
The method names (`setJobDurationMetricInHostMonitorWithoutMonitorType` and `setJobDurationMetricInHostMonitorWithMonitorType`) are overly verbose. `setJobDurationMetric` is already sufficiently descriptive, especially when supplemented by clear Javadoc that explains the method's behavior.